packages/firestore/test/lite/integration.test.ts

```diff
@@ -2038,3 +2045,142 @@
     });
   });
 });
+
+describe('countQuery()', () => {
+  const converter = {
+    toFirestore(input: { id: string; content: string }): DocumentData {
+      return { key: input.id, value: input.content };
+    },
+    fromFirestore(snapshot: QueryDocumentSnapshot): {
+      id: string;
+      content: string;
+    } {
+      const data = snapshot.data();
+      return { content: data.value, id: data.documentId };
+    }
+  };
+
+  const testDocs = [
+    { key: 'a', value: 'Adam' },
+    { key: 'b', value: 'Bob' },
+    { key: 'c', value: 'Chris' },
+    { key: 'a', value: 'Anna' },
+    { key: 'b', value: 'Bill' },
+    { key: 'c', value: 'Cooper' }
+  ];
+
+  it('empty collection count equals to 0', () => {
+    return withTestCollection(async coll => {
+      const countQuery_ = countQuery(query(coll));
+      expect(countQuery_.type).to.equal('AggregateQuery');
+
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(0);
+    });
+  });
+
+  it('test collection count equals to 6', () => {
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const countQuery_ = countQuery(query(collection));
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(6);
+    });
+  });
+
+  it('test collection count with filter', () => {
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_ = query(collection, where('key', '==', 'a'));
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(2);
+    });
+  });
+
+  it('test collection count with filter effected by small limit', () => {
+    // limit that is less that the actual count would work like count up_to.
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_ = query(collection, where('key', '==', 'a'), limit(1));
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(1);
+    });
+  });
+
+  it('test collection count with filter not effected by large limit', () => {
+    //limit that is larger than actual count wouldn't impact the return value
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_ = query(collection, where('key', '==', 'a'), limit(3));
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(2);
+    });
+  });
+
+  it('test collection count with converter on query', () => {
```

Comment:
Please remove this comment too, as it is just repeating what the test description already conveys. Also on line 2136.